src/interface/variable_state.hpp

```diff
@@ -0,0 +1,60 @@
+//========================================================================================
+// (C) (or copyright) 2020-2022. Triad National Security, LLC. All rights reserved.
+//
+// This program was produced under U.S. Government contract 89233218CNA000001 for Los
+// Alamos National Laboratory (LANL), which is operated by Triad National Security, LLC
+// for the U.S. Department of Energy/National Nuclear Security Administration. All rights
+// in the program are reserved by Triad National Security, LLC, and the U.S. Department
+// of Energy/National Nuclear Security Administration. The Government is granted for
+// itself and others acting on its behalf a nonexclusive, paid-up, irrevocable worldwide
+// license in this material to reproduce, prepare derivative works, distribute copies to
+// the public, perform publicly and display publicly, and to permit others to do so.
+//========================================================================================
+#ifndef INTERFACE_VARIABLE_STATE_HPP_
+#define INTERFACE_VARIABLE_STATE_HPP_
+
+#include <limits>
+
+#include "basic_types.hpp"
+#include "globals.hpp"
+#include "interface/metadata.hpp"
+#include "parthenon_arrays.hpp"
+
+namespace parthenon {
+
+static constexpr int InvalidSparseID = std::numeric_limits<int>::min();
+
+struct VariableState : public empty_state_t {
```

Comment:
will variable state always look the same? Should we name this something like `SparseState` in case we eventually have other kinds of state? I'm not sure this matters much and I'm fine leaving it as is.

Reply:
I think that we would probably add other non-sparse state to this object. This is the state that is carried around by all the data held by `CellVariable`s, whether or not they are sparse.